src/qt/transactionrecord.cpp

```diff
@@ -229,6 +229,12 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
                 // Mint to self
                 parts.append(TransactionRecord(hash, nTime, TransactionRecord::Anonymize, "",
                     -(nDebit - nChange), 0));
+            } else if (wtx.tx->IsSparkMint()) {
+                parts.append(TransactionRecord(hash, nTime, TransactionRecord::MintSparkToSelf, "",
+                    -(nDebit - nChange), 0));
+            } else if (wtx.tx->IsSparkSpend()) {
```

Comment:
You need to get nTxFee for spark spend transaction like it is done for lelantus joinsplit, at line 67, as spark amounts are shielded, you are unable to calculate fee by inputs - outputs.

Reply:
Updated. Thanks!